app/controllers/edit_directory_columns_controller.rb

```diff
@@ -26,16 +24,30 @@ def update
       raise Discourse::InvalidParameters, "Must have at least one column enabled"
     end
 
+    changes = "\n"
     directory_column_params[:directory_columns].values.each do |column_data|
       existing_column = directory_columns.detect { |c| c.id == column_data[:id].to_i }
       if (
-           existing_column.enabled != column_data[:enabled] ||
+           existing_column.enabled != ActiveModel::Type::Boolean.new.cast(column_data[:enabled]) ||
              existing_column.position != column_data[:position].to_i
          )
-        existing_column.update(enabled: column_data[:enabled], position: column_data[:position])
+        changes +=
+          "#{existing_column.name} - enabled: #{existing_column.enabled} -> #{column_data[:enabled]} , position: #{existing_column.position} -> #{column_data[:position]}\n"
```

Comment:
Hmm, we need to make sure that any text displayed to admins is localizable. The "enabled: " string here isn't localizable, and I think it's hard to explain this to translators as well. 


I see that `log_custom` doesn't let us use `previous_value` and `new_value` here, that's probably, so maybe we need a `log_user_column_change` entry after all. Then we can just pass the raw previous/new values.